src/components/PresetSection/presetSectionApi.ts

```diff
@@ -1,9 +1,48 @@
 import axios from "axios";
+import { ImportData } from "../../types/import-data";
+import itemData from "../../data/sorted_items.json";
 
 const apiUrl =
   "https://us-central1-pvmebackend.cloudfunctions.net/getPreset?id=";
 
 export const GetPreset = async (id: string) => {
   const response = await axios.get(`${apiUrl}${id}`);
-  return response.data;
+  const storedPreset = response.data;
+  let unpackedPreset = await unpackData(storedPreset);
+  return unpackedPreset;
+};
+
+const unpackData = async(stored: {
+  equipmentSlots: any; presetName: any; inventorySlots: string | any[]; 
+}): Promise<ImportData> => {
+  let newPreset: ImportData = {
+    presetName: "",
+    inventorySlots: [],
+    equipmentSlots: []
+  };
+  //create a map of all item labels to their default object
+  let itemDataMap = new Map();
+  await itemData.forEach(element => {
+    itemDataMap.set(element.label, element);
+  });
+  //loop stored preset inventory items
+  for(let i = 0 ; i < stored.inventorySlots.length ; i++) {
```

Comment:
btw if you install ESLint/Prettier it'll auto-fix formatting on save (e.g. adding space after `for`, `if`)

Reply:
weird i have both installed